File: assessments/object_relations/app/models/customer.rb
class Customer
  attr_accessor :name, :reviews, :restaurants
    @@customers = []
    def initialize(name)
      @name = name
      @reviews = []
      @@customers << self
    end
    def self.all
      @@customers
    end
    def self.find_or_create_by_name(name)
      customer = all.find { |customer| customer.name == name}
      if customer
        customer
      else
        Customer.new(name)
      end
    end
    def add_review(review, restaurant)
      review = Review.new(restaurant, review)
      @reviews << review
      review.customer = self
      review.restaurant = restaurant
      restaurant.reviews << review
      restaurant.customers << self
    end
    
end

# customer = customer.add_review('it was good', Restaurant.new)
# a new review tied ot the restaurant
# and tied to the customer

File: assessments/javascript/models/customer.js
const Store = {customers: [], restaurants: [], reviews: []}
class Customer {
  constructor(name) {
    this.name = name
    this.restaurants = []
    this.reviews = []
    Store.customers.push(this)
  }
  add_review(restaurant, review) {
    this.add_restaurant(restaurant)
    var rev = new Review(review, restaurant, this.name)
    this.reviews.push(rev)
    restaurant.reviews.push(rev)
    Store.reviews.push(rev)

  }
  add_restaurant(restaurant) {
    this.restaurants.push(restaurant)
    restaurant.customers.push(this)
    Store.restaurants.push(this)
  }
}